numba_rvsdg/tests/test_transforms.py

```diff
@@ -136,31 +190,50 @@ def test_dual_predecessor_and_dual_successor(self):
 
     def test_dual_predecessor_and_dual_successor_with_additional_arcs(self):
         original = """
-        "0":
-            jt: ["1", "2"]
-        "1":
-            jt: ["3"]
-        "2":
-            jt: ["1", "4"]
-        "3":
-            jt: ["0"]
-        "4":
-            jt: []
+        blocks:
+            '0':
+                type: basic
+            '1':
+                type: basic
+            '2':
+                type: basic
+            '3':
+                type: basic
+            '4':
+                type: basic
+        edges:
+            '0': ['1', '2']
+            '1': ['3']
+            '2': ['1', '4']
+            '3': ['0']
+            '4': []
+        backedges:
```

Comment:
I think this is wrong. The original should never have any `backedges`. They are just edges to begin with. Part of the SCFG loop restructure is to go through and make the distinction between edges and backedges.

Reply:
The issue is, if there's an edge (that is not a back edge) pointing towards the head of the graph, then we have no way of automatically detecting which node is the head of the graph from YAML. 

Since our code logic behaves as follows: 
The head of the graph is the node towards which no edge is pointing towards. 

To resolve this issue for this example we can simply have a placeholder head with no edges pointing towards it that in turn points towards the actual graph we need, that'll remove the need for back edges.

Reply:
This is also an offshoot variant of the issue faced by @sklam downstream in Numba, wherein he circumvented it by having a separate placeholder head and tail for every `RegionBlock`. Seems like that could be upstreamed here to fix it in the long run.

Reply:
I am not convinced that this is a good implementation. I think it should be possible to specify a cyclic graph in YAML and read it in. I think I can see a way to eliminate the need to detect a head. In general, I think cyclic, but reducible graphs should be supported.